master/main.go

```diff
@@ -11,43 +11,83 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/doriandekoning/IN4392-cloud-computing-lab/graphs"
 	"github.com/doriandekoning/IN4392-cloud-computing-lab/middleware"
 	"github.com/doriandekoning/IN4392-cloud-computing-lab/util"
+	"github.com/gorilla/mux"
 	"github.com/levigross/grequests"
 	uuid "github.com/satori/go.uuid"
-
-	"github.com/gorilla/mux"
 )
 
 var g graphs.Graph
 
 type worker struct {
-	Address string
-	Healty  bool
+	Address    string
+	InstanceId string
+	Healty     bool
+	Active     bool
 }
 
 var workers []*worker
 
+var Sess *session.Session
+
+const maxWorkers = 3
+const minWorkers = 1
+
+var requestCount = 0
+
 func main() {
 
+	var err error
+
+	Sess, err = session.NewSession(&aws.Config{
+		Region: aws.String("us-east-1"),
+	})
+
+	if err != nil {
+		log.Fatal("Error", err)
+	}
+
 	router := mux.NewRouter()
 	router.Use(middleware.LoggingMiddleWare)
 	router.HandleFunc("/health", GetHealth).Methods("GET")
+	router.HandleFunc("/killworkers", KillWorkersRequest).Methods("GET")
+	router.HandleFunc("/addworker", AddWorkerRequest).Methods("GET")
 	router.HandleFunc("/processgraph", ProcessGraph).Methods("POST")
 	router.HandleFunc("/worker/register", registerWorker).Methods("POST")
 	router.HandleFunc("/worker/unregister", unregisterWorker).Methods("DELETE")
 
+	go scaleWorkers()
 	go getWorkersHealth()
 	log.Fatal(http.ListenAndServe(":8000", router))
 
 }
 
 func GetHealth(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintln(w, "Registered workers: "+strconv.Itoa(len(workers))+"  Requests in the last minute: "+strconv.Itoa(requestCount))
+	for _, worker := range workers {
+		fmt.Fprintln(w, worker.Address+" -- "+worker.InstanceId+" -- "+strconv.FormatBool(worker.Healty))
+	}
+}
 
+func KillWorkersRequest(w http.ResponseWriter, r *http.Request) {
+	err := TerminateWorkers(workers)
+	if err == nil {
+		workers = nil
+		fmt.Println("Workers terminated")
+	}
+}
+
+func AddWorkerRequest(w http.ResponseWriter, r *http.Request) {
+	StartNewWorker()
+	fmt.Println("New worker started")
 }
 
 func ProcessGraph(w http.ResponseWriter, r *http.Request) {
+	requestCount++
```

Comment:
Misschien is een meer descriptive variabele naam beter, misschien iets als `requestsSinceScaling` oid.